### ISSUE TITLE
Provide guidance to language SIGs when they need to implement Event API

### ISSUE THREAD

**Issue by lmolkova** [open-telemetry/opentelemetry-specification#4193]
Spec currently says

https://github.com/open-telemetry/opentelemetry-specification/blob/0b3328bb17fa6f86af521f8d22dbbcddea2ac914/specification/logs/event-api.md?plain=1#L40-L53

also 

https://github.com/open-telemetry/opentelemetry-specification/blob/0b3328bb17fa6f86af521f8d22dbbcddea2ac914/specification/logs/event-api.md?plain=1#L79-L81

I believe it means that  some languages with common logging libraries don't need Event API at all and it's optional.

E.g. today I can write something like 

(.NET)

```csharp```
logger.LogInformation(new EventId(id: 42, name: "my-unique-event-name"), "something important");
```

(Java with slf4j):

```java
logger.atInfo()
 .addKeyValue("event.name", "com.foo.my-event-id")
 .log("something important") 
```

(Python)

```python
logger.error("something important", extra={"event.name": "com.foo.my-event-name"})
```

Providing structured event bodies (same as log bodies) is a bit more complicated and nuanced, but it seems we need to support it for both: events and logs.

- Presumably many languages can support events though existing common logging facades.
- In such languages Event-specific APIs (`EventLogger` `EventLoggerProvider`) are not necessary and don't need to be implemented. 
- Arguably `AnyValue` type (when necessary) is not specific to events and it should be possible to create a log record without `event.name` with complex body when logging API allows it

---------------------------
So, instead of implementing `EventLogger` and provider such languages could instead define and document how to use existing logging API to create events.

It seems like Event API should only be added when there is no logging API or when it does not provide reasonable ways to emit rich logs/events.

Related: https://github.com/open-telemetry/opentelemetry-specification/issues/4189

**Comment by jack-berg** [comment]
> Providing structured event bodies is possible too - it's a bit more complicated and nuanced

This is the key question to me. Even if its possible to record structured bodies, is it intuitive how to do so? At what point are the ergonomics poor enough to justify an API? 

I agree with the sentiment of this though.. if a logging API is available and has sufficiently good ergonomics, then using an OpenTelemetry event API only stands to complicate things.

A few other examples of impedance mismatch that may justify a event API:
- Obvious difference between body and attributes, which popular/common logging APIs likely don't make
- Ability to explicitly set the context to control whether or not the log is associated with a span
- Difference notion of scope (i.e. no scope attributes)
- ..maybe more? These examples were off the cuff.

**Comment by lmolkova** [comment]
I agree that there are caveats and uncertainties. I assume language SIG would decide whether there is a reasonable way to use logging API. 

I'd like them to evaluate options and make a decision.

**Comment by lmolkova** [comment]
To demonstrate that bodies have reasonable ergonomics at least in some languages:

Python: (supported today)

```python
logger.info({"key": "value", "complex_key": "complex_value"}, extra={"event.name": "com.foo.my-event-name"})
```

.NET (it does not map complex object to log body, but could) - see https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/logs/complex-objects/Program.cs#L35

```csharp
logger.FoodRecallNotice(new FoodRecallNotice
{
    BrandName = "Contoso",
    ProductDescription = "Salads",
    ProductType = "Food & Beverages",
    RecallReasonDescription = "due to a possible health risk from Listeria monocytogenes",
    CompanyName = "Contoso Fresh Vegetables, Inc.",
};);

internal static partial class LoggerExtensions
{
    [LoggerMessage(LogLevel.Critical)]
    public static partial void FoodRecallNotice(
        this ILogger logger,
        [LogProperties(OmitReferenceName = true)] in FoodRecallNotice foodRecallNotice);
}
```

**Comment by lmolkova** [comment]
I believe it could be closed now - we merged Event and Log API
